beginner_source/quickstart/optimization_tutorial.py

```diff
@@ -0,0 +1,197 @@
+        # move data to the device we use for computations
+        test_inputs, test_labels = 
+           test_inputs.to(device), test_labels.to(device)
+        pred = model(test_inputs)   # evaluate model on test minibatch
+        test_loss += cost_function(pred, test_labels).item() # compute loss
+        # compute the metrics for classification: 
+        # how many classes were guessed correctly
+        correct += 
+          (pred.argmax(1) == test_labels.argmax(1))
+          .type(torch.float).sum().item()
+
+    test_loss /= len(test_dataloader.dataset)
+    correct /= len(test_dataloader.dataset)
+    print('Epoch {} test Error:'.format(epoch))
+    print('acc: {:>0.1f}%, avg loss: {:>8f}'.format(100*correct, test_loss))
+
+Creating Custom Cost Functions
+------------------------------
+
+In addition to the included PyTorch cost functions you can create your own custom cost functions as long as they are differentiable. Here is an example of custom Cross Entropy Loss implementation from the `Stanford CS230 <https://cs230.stanford.edu/blog/pytorch/#loss-function>`_ course:
+
+.. code-block:: Python
+
+    def myCrossEntropyLoss(outputs, labels):
+        batch_size = outputs.size()[0]
+        # compute the log of softmax values
+        outputs = F.log_softmax(outputs, dim=1)
+        # pick the values corresponding to the labels
+        outputs = outputs[range(batch_size), labels]
+        return -torch.sum(outputs)/num_examples
+
+It can be called just like the out of the box implementation above.
+
+.. code-block:: Python
+
+    loss = myCrossEntropyLoss(model_prediction, true_value)
+
+A more in depth explanation of PyTorch cost functions is outside the scope of the quickstart but you can learn more
+about the different common cost functions for deep learning in the PyTorch `documentation <https://pytorch.org/docs/stable/nn.html#loss-functions>`_.
+
+Using Train/Validation/Test Split to Optimize Hyperparameters
+-------------------------------------------------------------
+
+In our example, we have split the data between train and test datasets. However, as we mentioned above, different hyperparameters can yield different model performance. Thus, it makes sense to use a part of the dataset for **hyperparameter optimization**. In this case, we split the dataset into three parts:
+
+ * Training data
+ * Validation data, which is used inside optimization loop to determine the accuracy of the current model and the optimal number of epochs. After a certain number of epochs, validation accuracy typically starts to decrease, which means that we have reached optimal performance for given hyperparameters.
+ * Test data, which is used to measure the performance of the model for given hyperparameters. It is important that test data are independent from validation data, i.e. the same dataset cannot be used for both validation and test purposes.
+
+We will not consider hyperparameter optimization further in this quickstart. 
```

Comment:
This information is not strictly within the scope of our Quickstart. I have earlier linked to a hyperparameter tuning tutorial for interested folks.

```suggestion
```